pkg/server/api/agent/v1/service.go

```diff
@@ -47,7 +55,52 @@ func (s *Service) DeleteAgent(ctx context.Context, req *agent.DeleteAgentRequest
 }
 
 func (s *Service) BanAgent(ctx context.Context, req *agent.BanAgentRequest) (*empty.Empty, error) {
-	return nil, status.Error(codes.Unimplemented, "method not implemented")
+	log := rpccontext.Logger(ctx)
+
+	if req.Id == nil {
+		log.Error("Invalid request: ID is nil")
+		return nil, status.Error(codes.InvalidArgument, "ID is nil")
+	}
+
+	id, err := spiffeid.New(req.Id.TrustDomain, req.Id.Path)
+	if err != nil {
+		log.WithError(err).Error("Invalid request: invalid SPIFFE ID")
+		return nil, status.Errorf(codes.InvalidArgument, "invalid SPIFFE ID: %v", err)
+	}
+
+	log = log.WithField(telemetry.SPIFFEID, id.String())
+
+	if !strings.Contains(id.Path(), "spire/agent/") {
+		log.Error("Invalid request: ID is not an agent SPIFFE ID")
```

Comment:
how about:
```
invalid SPIFFE ID: not an agent ID
```

Reply:
For previous RPCs, we agreed to prefix errors related to invalid input arguments with "Invalid request". So, perhaps: `Invalid request: not an agent ID`?